app/src/org/commcare/network/HttpRequestGenerator.java

```diff
@@ -160,10 +165,12 @@ public HttpResponse makeCaseFetchRequest(String baseUri, boolean includeStateFla
 
         String uri = serverUri.toString();
         Log.d(TAG, "Fetching from: " + uri);
-        HttpGet request = new HttpGet(uri);
-        AndroidHttpClient.modifyRequestToAcceptGzipResponse(request);
-        addHeaders(request, syncToken);
-        return execute(client, request);
+        currentRequest = new HttpGet(uri);
+        AndroidHttpClient.modifyRequestToAcceptGzipResponse(currentRequest);
+        addHeaders(currentRequest, syncToken);
+        HttpResponse response = execute(client, currentRequest);
+        currentRequest = null;
+        return response;

```

Comment:
Only setting the `currentResponse` in `makeCaseFetchRequest`, because we don't care about aborting other types of http requests (yet)